### DISCUSSION TITLE
Install package from USB

### DISCUSSION THREAD

**Discussion by redboyhun** [procount/pinn#645] [Q&A] [answered]
Based on the description I should be able to install tar/tar.xz packages from USB drive.
I tried both:
- place tar and tar.xz packages under root of the drive
- place tar and tar.xz packages under /os directory of the drive

In both cases I could not see them in the list. USB was formatted as FAT32 and saw it as possible destination for installations.
Any other special requirement?

**Comment by procount** [comment, marked as answer]
Matt didn't give you the whole picture. The tar.xz file is the only _additional_ file you need. You still need all the additional files to describe the OS to PINN.
You need to take a copy of the whole of the PINN Batocera folder and replace the tar.xz file. Ideally you should update the json files too, and if there are any checksums, you probably need to update or delete them.
You need to put the folder under the /os/ folder on your FAT32 formatted USB stick. The name of the folder is not important since the OS name comes from the os.json file, Without that file, the OS will not even be shown. Please also see https://github.com/procount/pinn/blob/master/README_PINN.md#how-to-create-a-custom-os-version got more information
